lib/src/graph.rs

```diff
@@ -148,10 +154,12 @@ impl<N> Default for TopoGroupedGraphNode<N> {
     }
 }
 
+type GraphEdgeResult<N> = Result<Option<(N, Vec<GraphEdge<N>>)>, RevsetEvaluationError>;
+
 impl<N, I> TopoGroupedGraphIterator<N, I>
 where
     N: Clone + Hash + Eq,
-    I: Iterator<Item = (N, Vec<GraphEdge<N>>)>,
+    I: RevsetIterator<(N, Vec<GraphEdge<N>>)>,
```

Comment:
`TopoGroupedGraphIterator` is used in other contexts, so the error type would have to be generic. For example, `show_op_diff()` can use `E = Infallible`.